File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/transactions/db/TransactionPersistanceDBModule.java
package com.fintechplatform.ui.transactions.db;

import com.fintechplatform.ui.money.MoneyHelper;
import com.fintechplatform.ui.transactions.models.TransactionDetailHelper;
import com.fintechplatform.ui.transactions.models.TransactionHelper;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class TransactionPersistanceDBModule {

    @Provides
    @Singleton
    TransactionPersistanceDB providesTransactionsPersistanceDB(TransactionDB db, TransactionHelper helper) {
        return new TransactionPersistanceDB(db, helper);
    }

    @Provides
    @Singleton
    TransactionDB providesTransactionDB() {
        return new TransactionDB();
    }

    @Provides
    @Singleton
    TransactionHelper providesTransactionHelper(MoneyHelper moneyHelper) {
        return new TransactionHelper(moneyHelper);
    }

    @Provides
    @Singleton
    TransactionDetailHelper providesTransactionDetailHelper() {
        return new TransactionDetailHelper();
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/transactions/di/TransactionPersistanceDBModule.kt
package com.fintechplatform.ui.transactions.di

import com.fintechplatform.ui.money.MoneyHelper
import com.fintechplatform.ui.transactions.db.TransactionDB
import com.fintechplatform.ui.transactions.db.TransactionPersistanceDB
import com.fintechplatform.ui.transactions.models.TransactionDetailHelper
import com.fintechplatform.ui.transactions.models.TransactionHelper
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class TransactionPersistanceDBModule {

    @Provides
    @Singleton
    internal fun providesTransactionsPersistanceDB(db: TransactionDB, helper: TransactionHelper): TransactionPersistanceDB {
        return TransactionPersistanceDB(db, helper)
    }

    @Provides
    @Singleton
    internal fun providesTransactionDB(): TransactionDB {
        return TransactionDB()
    }

    @Provides
    @Singleton
    internal fun providesTransactionHelper(moneyHelper: MoneyHelper): TransactionHelper {
        return TransactionHelper(moneyHelper)
    }

    @Provides
    @Singleton
    internal fun providesTransactionDetailHelper(): TransactionDetailHelper {
        return TransactionDetailHelper()
    }
}
